File: DSA_pattern/HashMap/findsecondnonduplicate.js
let value = 'aaabcdbcffvvgee'

let hashmap = {}

for(i=0;i < value.length;i++){
    hashmap[value[i]] = (hashmap[value[i]] || 0) + 1
}

let count = 0

for(const key in hashmap){
    if(hashmap[key] == 1){
        count += 1
        if(count == 2){
            console.log(key)
        }
    }
}
   
        

console.log(hashmap)

File: DSA_pattern/HashMap/findsecondnonduplicate.py
value = 'aaabcdbcffvvgee'

def findsecondnonduplicate(value):
    hashmap = {}
    for i in range(len(value)):
        hashmap[value[i]] = hashmap.get(value[i], 0) + 1
    
    print(hashmap)
    count = 0
    for i in range(len(value)):
        if hashmap[value[i]] == 1:
            count += 1
            if count == 2:
                return value[i]
    return -1

print(findsecondnonduplicate(value))